C-Unit1-assignments/src/Assignment_02.cpp

```diff
@@ -0,0 +1,96 @@
+
+//
+// Compute factorial of a given positive integer
+// 0 <= n <= 100
+//
+// return value         - number of digits in n!
+// factorialDigits[158] - digits of n!
+//
+
+int digits(int n) {
+	//printf("%d di \n", n);
+	int l = 0;
+	while (n) {
+		n /= 10;
+		l++;
+	}
+	return l;
+}
+void shift(int n, int l, int factorial[]) {
+	//printf("shift %d %d \n", n, l);
+	for (int i = l - 1;i >= 0;i--) {
+		factorial[i + n] = factorial[i];
+	}
+}
+int factorialOfN(int n, int factorialDigits[158]) {
+	for (int k = 0;k < 158;k++)factorialDigits[k] = 0;
+	factorialDigits[0] = 1;
+	if (n == 0 || n == 1)return 1;
+	int l = 0;
+	int i = 0;
+	int j;
+	int p = 0, c = 0;
+	while (n) {
+		i = l;
+		while (i >= 0) {
```

Comment:
Even this one. In the worst case, it might go upto O(n^2) to O(n^3). Can you try to optimize this. You discuss with @sritej or @manoj regarding this. They have already done this. Also make it modular.